edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -305,10 +315,7 @@ async Task<bool> DeleteMessageCallback(long offset, MessageRef messageRef)
 
                                 if (deleteMessage)
                                 {
-                                    if (checkpointData.Offset < offset && enqueuedTime >= messageRef.TimeToLive)
```

Comment:
>if (checkpointData.Offset < offset && enqueuedTime >= messageRef.TimeToLive) [](start = 36, length = 76)

Seems like this is needed?  We only want to increment the counter here for messages that were not sent but have expired.

Reply:
Good catch. Fixed.